lib/src/Uncertainty/Bayesian/IndependentMetropolisHastings.cxx

```diff
@@ -0,0 +1,130 @@
+/**
+ *  @brief Metropolis-Hastings algorithm
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/IndependentMetropolisHastings.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+CLASSNAMEINIT(IndependentMetropolisHastings)
+
+static const Factory<IndependentMetropolisHastings> Factory_IndependentMetropolisHastings;
+
+/* Default constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings()
+  : MetropolisHastingsImplementation()
+{
+  // Nothing to do
+}
+
+
+/* Parameters constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings(const Distribution & targetDistribution,
+                                                          const Point & initialState,
+                                                          const Distribution & proposal,
+                                                           const Indices & marginalIndices)
+  : MetropolisHastingsImplementation(targetDistribution, initialState, marginalIndices)
+{
+  setProposal(proposal);
+}
+
+/* Parameters constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings(const Function & targetLogPDF,
+                                                          const Domain & support,
+                                                          const Point & initialState,
+                                                          const Distribution & proposal,
+                                                           const Indices & marginalIndices)
+  : MetropolisHastingsImplementation(targetLogPDF, support, initialState, marginalIndices)
+{
+  setProposal(proposal);
+}
+
+
+/* String converter */
+String IndependentMetropolisHastings::__repr__() const
+{
+  return OSS() << "class=" << IndependentMetropolisHastings::GetClassName()
+         << " name=" << getName()
+         << " derived from " << MetropolisHastingsImplementation::__repr__()
+         << " proposal=" << proposal_;
+}
+
+
+IndependentMetropolisHastings* IndependentMetropolisHastings::clone() const
+{
+  return new IndependentMetropolisHastings(*this);
+}
+
+
+Point IndependentMetropolisHastings::getCandidate() const
+{
+  const Point candidate(proposal_.getRealization());
```

Comment:
you still need to return the candidate variable and not another realization

Reply:
Thank you Julien for finding this bug... Joseph has fixed it.